targets/psa_switch/psa_switch.cpp

```diff
@@ -316,8 +316,24 @@ PsaSwitch::ingress_thread() {
     Parser *parser = this->get_parser("ingress_parser");
     parser->parse(packet.get());
 
-    // set default std metadata values
+    // passing metadata
+    phv->get_field("psa_ingress_input_metadata.ingress_port").set(ingress_port);
+
+    phv->get_field("psa_ingress_input_metadata.packet_path").set(
+        phv->get_field("psa_ingress_parser_input_metadata.packet_path"));
+
+    phv->get_field("psa_ingress_input_metadata.parser_error").set(
+        packet->get_error_code().get());
+
+    phv->get_field("psa_ingress_input_metadata.ingress_timestamp")
+        .set(get_ts().count());
```

Comment:
The PSA spec actually recommends setting this timestamp to the time that the packet began the ingress parser, so perhaps a little bit better to save the value of `get_ts().count` in a local variable just before parsing, then assign this value here.